vars/buildDockerImage.groovy

```diff
@@ -9,36 +9,47 @@ void call(Map args = [:]) {
     String x64TarGz = inputManifest.getPublicDistUrl("${PUBLIC_ARTIFACT_URL}", "${JOB_NAME}", "${BUILD_NUMBER}", 'linux', 'x64')
     String arm64TarGz = inputManifest.getPublicDistUrl("${PUBLIC_ARTIFACT_URL}", "${JOB_NAME}", "${BUILD_NUMBER}", 'linux', 'arm64')
 
-    def parameters = [
-        string(name: 'DOCKER_BUILD_GIT_REPOSITORY', value: 'https://github.com/opensearch-project/opensearch-build'),
-        string(name: 'DOCKER_BUILD_GIT_REPOSITORY_REFERENCE', value: 'main'),
-        string(name: 'DOCKER_BUILD_SCRIPT_WITH_COMMANDS', value: [
-                'id',
-                'pwd',
-                'cd docker/release',
-                "curl -sSL ${x64TarGz} -o ${filename}-x64.tgz",
-                "curl -sSL ${arm64TarGz} -o ${filename}-arm64.tgz",
-                [
-                    'bash',
-                    'build-image-multi-arch.sh',
-                    "-v ${inputManifest.build.version}",
-                    "-f ./dockerfiles/${filename}.al2.dockerfile",
-                    "-p ${filename}",
-                    "-a 'x64,arm64'",
-                    "-r ${filename}staging/${filename}",
-                    "-t '${filename}-x64.tgz,${filename}-arm64.tgz'",
-                    "-n ${BUILD_NUMBER}"
-                ].join(' ')
-            ].join(' && ')),
-        booleanParam(name: 'IS_STAGING', value: true)
-    ]
+    def x64Response = args.dryRun ? [status: 200] : httpRequest(httpMode: 'HEAD', url: x64TarGz)
+    def arm64Response = args.dryRun ? [status: 200] : httpRequest(httpMode: 'HEAD', url: arm64TarGz)
 
-    if (! args.dryRun) {
-        dockerBuild: {
-            build job: 'docker-build',
-            parameters: parameters
-        }
+    if (x64Response.status != 200 || arm64Response.status != 200) {
+        echo([
+            'Skipping docker build.',
```

Comment:
Seems like this should error out if the artifacts cannot be found.

Reply:
You're right, but there's no other easy way to tell the docker build that the incremental build didn't build without introducing a global, and I couldn't get it to work in tests.